### PR TITLE
Makes incorporeal things properly immune to bonfires and lava (and tram and elevator...)

### PR DESCRIPTION
**Pull request by Heroman3003**
Shadekin suffrage lessens

### CODE REVIEW

**Comment by Very-Soft** [comment]
\o/ praise

**Comment by ResidentCody** [comment]
\o/ praise

File: code/game/turfs/simulated/lava.dm [hunk 1]
```diff
@@ -95,6 +95,6 @@
 
 // Tells AI mobs to not suicide by pathing into lava if it would hurt them.
 /turf/simulated/floor/lava/is_safe_to_enter(mob/living/L)
-	if(!is_safe() && !L.hovering)
+	if(!is_safe() && !L.hovering && !(L.is_incorporeal()))
```
Comment (Arokha) [inline code/game/turfs/simulated/lava.dm:98]: This won't actually make them immune to lava. You'd wanna do that down in burn_stuff()

Comment (Heroman3003) [reply code/game/turfs/simulated/lava.dm]: Okay, I kind of knew that but when actually doing changes I was for some reason 100% certain proc i was modifying was is_safe... despite the fact that is_safe is referenced in that very check. Either way. Fixed.

**Comment by Screemonster** [comment]
yesssssssssssssssssssssssssss

**Comment by Enzo-Leon** [comment]
HAH finally! Good to see!

**Pull request merged**